src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -787,4 +787,24 @@ object messages {
     val msg = "unreachable code"
     val explanation = ""
   }
+
+  case class IncorrectSequenceWildcardPatternPosition()(implicit ctx: Context) extends Message(31) {
+    val kind = "Syntax"
+    val msg = "`_*' can be used only for last argument"
+    val explanation =
+      hl"""|Sequence wildcard pattern is expected at the end of an argument list.
+           |This pattern matches any remaining elements in a sequence.
+           |Consider the following example:
+           |
+           |def sumOfTheFirstTwo(list: List[Int]): Int = list match {
+           |  case List(first, second, x:_*) => first + second
+           |  case _ => 0
+           |}
```

Comment:
To get highlighting on this part pull it out to a variable like:

```scala
val code = "def sumOfThe..."

hl"""|Sequence wildcard pattern is expected at the end of an argument list.
     |This pattern matches any remaining elements in a sequence.
     |Consider the following example:
     |
     |$code
     |
     |..."""
```

And for the code under `Calling:`

interpolate it like:

```
|Calling:
|
|${"sumOfTheFirstTwo(List(1, 2, 10))"}
|
|would give 3 as a result"""
```

this will highlight the code.